servlet/src/main/java/io/undertow/servlet/core/ManagedServlet.java

```diff
@@ -104,7 +104,9 @@ public void setupMultipart(ServletContextImpl servletContext) {
                 if(locFile.isAbsolute()) {
                     tempDir = locFile;
                 } else {
-                    tempDir = servletContext.getDeployment().getDeploymentInfo().getTempPath().resolve(location);
+                    final Path baseTmpDir = servletContext.getDeployment().getDeploymentInfo().getTempPath();
+                    tempDir = baseTmpDir == null ? Paths.get(System.getProperty("java.io.tmpdir"), location)
+                            : baseTmpDir.resolve(location);
```

Comment:
DeploymentManagerImpl uses `SecurityActions.getSystemProperty("java.io.tmpdir")`, I think we should use that utility here as well.

This issue has come up a few times in several different places, I wonder if we should initialize `DeploymentInfo.tempDir` with default value `SecurityActions.getSystemProperty("java.io.tmpdir")` instead of spot-fixing each time it's used elsewhere?

Reply:
Sorry, I have been away, so couldn't reply earlier.
>> DeploymentManagerImpl uses SecurityActions.getSystemProperty("java.io.tmpdir"), I think we should use that utility here as well.

You are right, that needs to be done here too. I'll update accordingly.

>>This issue has come up a few times in several different places, I wonder if we should initialize DeploymentInfo.tempDir with default value SecurityActions.getSystemProperty("java.io.tmpdir") instead of spot-fixing each time it's used elsewhere?

The thing is, the DeploymenInfo has a setter which allows null value to be set to this member, which means that setting a default value won't solve this issue fully (some code could actually set a null value to overwrite this default value). I have an alternate solution in mind, which I will include in the update to this PR. Thank you @carterkozak for this review.